File: problem_solutions/CSES/MissingNumber.py
n = int(input())

nums = [int(n) for n in input().split(",")]

# using the sum method

actual_sum = n*(n+1)//2
given_sum = sum(nums)

missing_num = actual_sum-given_sum

print(missing_num)

# using the XOR method

xor = 0

for i in range(n-1):
    xor^=nums[i]
    xor^=(i+1)

xor^=n
print(xor)

File: problem_solutions/CSES/MissingNumber.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> nums(n - 1);
    for (int i = 0; i < n - 1; i++) {
        cin >> nums[i];
    }

    // Using XOR method
    int xor_result = 0;

    for (int i = 0; i < n - 1; i++) {
        xor_result ^= nums[i];
        xor_result ^= (i + 1);
    }
    xor_result ^= n;

    cout << xor_result << endl;
    return 0;
}
